### PR TITLE
MRG: Fixes for #11090

### PR DESCRIPTION
**Pull request by hoechenberger**
@larsoner I discovered a bug (mapping temperature to Coulomb instead of Celsius) and an oversight in #11090 when trying to actually read GSR and temperature data.

### CODE REVIEW

File: mne/channels/channels.py [hunk 1]
```diff
@@ -202,7 +202,8 @@ def equalize_channels(instances, copy=True, verbose=None):
                FIFF.FIFF_UNIT_T_M: 'T/m',
                FIFF.FIFF_UNIT_MOL: 'M',
                FIFF.FIFF_UNIT_NONE: 'NA',
-               FIFF.FIFF_UNIT_CEL: 'C'}
+               FIFF.FIFF_UNIT_CEL: 'C',
```
Comment (hoechenberger) [inline mne/channels/channels.py:205]: Are we bound to a string without spaces and no special characters? Can we change this to "degrees C" or "°C"?

**Comment by agramfort** [comment]
let's stick to C to be consistent. It's the SI unit.

**Review by agramfort — APPROVED** [review]

**Comment by agramfort** [comment]
thx @hoechenberger

**Comment by hoechenberger** [comment]
> let's stick to C to be consistent. It's the SI unit.

Well no, it's not, that's why I'm asking because it's bugging me so much. The SI unit is degrees Celsius, °C

C is Coulomb. So using C for temperature is factually wrong

**Comment by agramfort** [comment]
BIDS says to use "oC"  see
https://bids-specification.readthedocs.io/en/stable/99-appendices/05-units.html

**Comment by hoechenberger** [comment]
> BIDS says to use "oC"  see
> https://bids-specification.readthedocs.io/en/stable/99-appendices/05-units.html
> 
> > Message ID: ***@***.***>
> >
> 

Omg, this has got to be the most unfortunate choice ... I thought one of the focus points of BIDS was on human-readable metadata ... I bet if I ask lab colleagues, none of them would know what "oC" is supposed to mean :( but it's in the spec now and I don't have the time and energy to push for a change, so this is just what we'll have to live with for BIDS

But I'd really not want to introduce this ... atrocity... into MNE 😅
If we cannot have the degree sign (can we not?), we could call the unit "degC". WDYT?

**Comment by larsoner** [comment]
We could try the unicode degree. I think we only really use this unit mapping for plotting, and for converting scale factors (e.g., `fT` to `T`). Both of those should work I think, maybe with some minor adjustments for the two characters in the units.

**Comment by agramfort** [comment]
@hoechenberger try and see if it breaks anything. Eg IO plotting etc.

**Pull request merged**